cmd/soroban-cli/src/log/event.rs

```diff
@@ -0,0 +1,50 @@
+use crate::xdr;
+
+mod log {
+    pub fn log(i: usize, event: &super::xdr::DiagnosticEvent) {
+        tracing::info!("{i}: {event:#?}");
+    }
+}
+
+mod contract {
+
+    pub fn contract(i: usize, event: &super::xdr::DiagnosticEvent) {
+        tracing::info!("{i}: {event:#?}");
+    }
+}
+
+mod diagnostic {
+    pub fn diagnostic(i: usize, event: &super::xdr::DiagnosticEvent) {
+        tracing::debug!("{i}: {event:#?}");
+    }
+}
+
+pub fn events(events: &[xdr::DiagnosticEvent]) {
+    for (i, event) in events.iter().enumerate() {
+        if is_contract_event(event) {
+            contract::contract(i, event);
+        } else if is_log_event(event) {
+            log::log(i, event);
+        } else {
+            diagnostic::diagnostic(i, event);
+        }
+    }
+}
+
+fn is_contract_event(event: &xdr::DiagnosticEvent) -> bool {
+    matches!(event.event.type_, xdr::ContractEventType::Contract)
+}
+
+fn is_log_event(event: &xdr::DiagnosticEvent) -> bool {
+    match &event.event.body {
+        xdr::ContractEventBody::V0(xdr::ContractEventV0 { topics, .. }) if topics.len() == 1 => {
```

Comment:
❗ We should probably constrain this to the `Diagnostic` event type too. Although I guess it doesn't matter so much because is_contract_event will catch any `log` topic contract event, but with these functions being separate they look like they're able to standalone when in reality they are tightly coupled on being used together and in a specific order.